reactive/docker.py

```diff
@@ -425,42 +408,29 @@ def install_cuda_drivers_repo(architecture, release, ubuntu):
     :param ubuntu: String
     :return: None
     """
-    key_file = '7fa2af80.pub'
+    key_file = "7fa2af80.pub"
 
     # Distribution should be something like 'ubuntu1604'.
-    distribution = '{}{}'.format(
-        ubuntu,
-        str(release).replace('.', '')
-    )
-    repository_path = \
-        'developer.download.nvidia.com/compute/cuda/repos/{}/x86_64'.format(
+    distribution = "{}{}".format(ubuntu, str(release).replace(".", ""))
+    repository_path = (
+        "developer.download.nvidia.com/compute/cuda/repos/{}/x86_64".format(
             distribution
         )
-    command = 'apt-key adv --fetch-keys http://{}/{}'.format(
-        repository_path,
-        key_file
     )
+    command = "apt-key adv --fetch-keys http://{}/{}".format(repository_path, key_file)
     check_call(split(command))
 
-    cuda_repository_version = config('cuda_repo')
-    cuda_repository_package = 'cuda-repo-{}_{}_{}.deb'.format(
-        distribution,
-        cuda_repository_version,
-        architecture
-    )
-    repository_url = 'https://{}/{}'.format(
-        repository_path,
-        cuda_repository_package
+    cuda_repository_version = config("cuda_repo")
+    cuda_repository_package = "cuda-repo-{}_{}_{}.deb".format(
+        distribution, cuda_repository_version, architecture
     )
+    repository_url = "https://{}/{}".format(repository_path, cuda_repository_package)
 
-    r = requests.get(repository_url)
-    r.raise_for_status()
-    with open(cuda_repository_package, 'wb') as f:
-        for chunk in r.iter_content(chunk_size=1024):
-            f.write(chunk)
-    r.close()
+    with contextlib.closing(urlopen(repository_url)) as r:
+        with open(cuda_repository_package, "wb") as f:
+            f.write(r.read())
```

Comment:
For those looking -- this is the only change to remove the requests package

Reply:
`urlopen(...)` will raise when there are protocol errors -- just like `r.raise_for_status()` did

Reply:
Just a quick suggestion here: I don't know exactly what the size of the package is; however, caution should be taken with `r.read()` for large files, as this can consume a significant amount of host memory.

Reply:
hmm... true it is pulling the whole deb package. Lemme fix that